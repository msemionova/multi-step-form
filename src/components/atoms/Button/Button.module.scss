.button {
  font: var(--typography-desktop-button-text);
  border-radius: var(--border-radius-medium);
  padding: 1.6rem 2.4rem;
  transition: background-color 200ms;

  &--primary {
    background-color: var(--color-primary-dark);
    color: var(--color-white);

    &:hover,
    &:active {
      background-color: var(--color-primary);
    }
  }

  &--transparent {
    background-color: transparent;
    color: var(--color-black-text);
    padding: 1.6rem 0;
    transition: color 200ms;

    &:hover,
    &:active {
      color: var(--color-primary-dark);
    }
  }
}
