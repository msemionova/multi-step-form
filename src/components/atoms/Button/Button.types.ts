export type ButtonProps = {
  children: React.ReactNode,
  onClick: React.MouseEventHandler<HTMLElement>,
  id?: string,
  type?: 'primary' | 'transparent'
}
