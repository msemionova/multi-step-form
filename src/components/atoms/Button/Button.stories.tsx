import type { Meta, StoryObj } from '@storybook/react';

import Button from './Button';

const meta: Meta<typeof Button> = {
  component: Button,
};

export default meta;
type Story = StoryObj<typeof Button>;

export const Primary: Story = {
  args: {
    children: 'Default button text'
  }
};

export const Transparent: Story = {
  args: {
    children: 'Default button text',
    type: 'transparent'
  }
};
