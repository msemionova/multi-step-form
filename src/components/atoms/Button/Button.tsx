import styles from './Button.module.scss';
import { ButtonProps } from './Button.types';

const Button = ({ children, onClick, id = '', type = 'primary' }: ButtonProps) => {
  const typeClass = styles[`button--${type}`];
  const classes = [styles['button'], typeClass].filter(Boolean).join(' ');
  const ariaLabel = id.replace('-', ' ');

  return (
    <>
      <button
        id={id}
        data-testid={id}
        onClick={onClick}
        aria-label={ariaLabel}
        className={classes}
      >
        <span>{children}</span>
      </button>
    </>
  );
};

export default Button;
