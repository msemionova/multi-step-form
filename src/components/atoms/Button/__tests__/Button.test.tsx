import Button from '../Button';
import renderer from 'react-test-renderer';
import { fireEvent, render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';

describe('<Button />', () => {
  const mockOnClick = jest.fn();
  const buttonText = 'Test Button';

  it('renders correctly', () => {
    const tree = renderer
      .create(<Button onClick={mockOnClick}>{buttonText}</Button>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly with customCssClass and secondary passed', () => {
    const tree = renderer
      .create(
        <Button
          onClick={mockOnClick}
          type='primary'
        >
          {buttonText}
        </Button>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should display the text passed as props', () => {
    const { getByText } = render(<Button onClick={mockOnClick}>{buttonText}</Button>);
    expect(getByText(buttonText)).toBeInTheDocument();
  });

  it('should trigger the passed function if clicked', () => {
    render(<Button id='button-test' onClick={mockOnClick}>{buttonText}</Button>);
    const element = screen.getByTestId('button-test');
    expect(mockOnClick).not.toBeCalled();
    fireEvent.click(element);
    expect(mockOnClick).toBeCalled();
  });
});
