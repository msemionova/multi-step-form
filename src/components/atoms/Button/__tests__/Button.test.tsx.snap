// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> renders correctly 1`] = `
<button
  aria-label=""
  className="button button--primary"
  data-testid=""
  id=""
  onClick={[MockFunction]}
>
  <span>
    Test Button
  </span>
</button>
`;

exports[`<Button /> renders correctly with customCssClass and secondary passed 1`] = `
<button
  aria-label=""
  className="button button--primary"
  data-testid=""
  id=""
  onClick={[MockFunction]}
>
  <span>
    Test Button
  </span>
</button>
`;
